jgrapht-core/src/main/java/org/jgrapht/alg/tour/TwoApproxMetricTSP.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.tour;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.jgrapht.GraphPath;
+import org.jgrapht.GraphTests;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.interfaces.TSPAlgorithm;
+import org.jgrapht.alg.spanning.KruskalMinimumSpanningTree;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.GraphWalk;
+import org.jgrapht.graph.SimpleGraph;
+import org.jgrapht.traverse.DepthFirstIterator;
+
+/**
+ * A 2-approximation algorithm for the metric TSP problem.
+ * 
+ * <p>
+ * The travelling salesman problem (TSP) asks the following question: "Given a list of cities and
+ * the distances between each pair of cities, what is the shortest possible route that visits each
+ * city exactly once and returns to the origin city?". In the metric TSP, the intercity distances
+ * satisfy the triangle inequality.
+ * 
+ * <p>
+ * This is an implementation of the folklore algorithm which returns a depth-first ordering of the
+ * minimum spanning tree. The algorithm is a 2-approximation assuming that the instance satisfies
+ * the triangle inequality. The running time is O(|V|^2 log|V|).
+ * 
+ * <p>
+ * See <a href="https://en.wikipedia.org/wiki/Travelling_salesman_problem">wikipedia</a> for more
+ * details.
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ */
+public class TwoApproxMetricTSP<V, E>
+    implements TSPAlgorithm<V, E>
+{
+    /**
+     * Construct a new instance
+     */
+    public TwoApproxMetricTSP()
+    {
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Could you specify that the input graph for this particular algorithm needs to be a complete graph. Not all TSP problems are defined on a complete graph.